django/core/management/commands/inspectdb.py

```diff
@@ -329,7 +329,15 @@ def get_field_type(self, connection, table_name, row):
 
         # Add max_length for all CharFields.
         if field_type == "CharField" and row.display_size:
-            field_params["max_length"] = int(row.display_size)
+            if (size := int(row.display_size)) and size > 0:
+                field_params["max_length"] = size
+            elif not connection.features.supports_unlimited_varchar:
+                field_params["max_length"] = 50
+                field_notes.append(
+                    "max_length has been guessed, as the "
+                    "database returned a negative internal size and does not "
+                    "support unlimited VARCHAR"
+                )
```

Comment:
IMO, we shouldn't be guessing and it's definitely no need to change that in this PR.